src/function/gds/output_writer.cpp

```diff
@@ -111,6 +113,10 @@ void PathsOutputWriter::write(processor::FactorizedTable& fTable, nodeID_t dstNo
             if (checkPathNodeMask(curPath) && checkSemantic(curPath)) {
                 writePath(curPath);
                 fTable.append(vectors);
+                numOutputTuples++;
+                if (numOutputTuples >= info.limitNumber) {
```

Comment:
As I said, i would probably use a global numOutputTuples atomic value in a shared state and have the write() function and the vertexCompute function related to output writers stop early, instead of doing a thread-level limit counting and then further aggregating. I think for limit, that does not seem necessary.

Reply:
Done